includes/Title.php

```diff
@@ -147,6 +147,11 @@ public static function newFromText( $text, $defaultNamespace = NS_MAIN ) {
 
 		$t = new Title();
 		$t->mDbkeyform = str_replace( ' ', '_', $filteredText );
+		# Prevent error when title ends with colon and title
+		# is interpreted as namespace:[empty title]
+		if ( substr($t -> mDbkeyform, -1 ) == ':' ) {
+			$t->mDbkeyform = substr($t -> mDbkeyform, 0, -1);
+		}
 		$t->mDefaultNamespace = $defaultNamespace;
```

Comment:
This method occurs in our code more than 1000 times. I'm not sure if changing its logic is safe. For sure it will break for articles which end with ":" or articles which contain only ":".